packages/frontend/model/extract-nav.ts

```diff
@@ -16,6 +16,24 @@ const getLink = (data: {}, { isPillar }: { isPillar: boolean }): LinkType => {
     };
 };
 
+const rrLinkConfig = 'config.readerRevenueLinks';
+const buildRRLinkCategories = (
+    data: {},
+    el: string,
+): ReaderRevenueCategories => ({
+    subscribe: getString(data, `${rrLinkConfig}.${el}.subscribe`, ''),
```

Comment:
@GHaberis @SiAdcock @AWare @nicl  This simplest of things got very confusing with Typescript (see my attempt at reducing, which just got too complex with TS https://github.com/guardian/dotcom-rendering/pull/389/commits/c8f01a7b287a0666ea55932c779bb535334f2c0d). Chatted to Alex and backed out to halfway house of manually typing - the only thing I don't like is that the string value isn't type checked (i.e how would I do 'subscribe' is one of the keys from this interface or use the key from the current object so someone couldn't change it to 'subscribed' or whatever? OR does it matter...)

This is more a learning exercise for me at this point and I think the data model restructure or a Type -> JSON conversion thingy-ma-bob is more of a wider solution.

Reply:
> one thing I don't like is that the string value isn't type checked (i.e how would I do 'subscribe' is one of the keys from this interface or use the key from the current object so someone couldn't change it to 'subscribed' or whatever? OR does it matter...

I'd say being able to access properties of the data sent from frontend with confidence does matter. Having an fully typed interface that represents the data from frontend would give us that confidence, and ideally we'd be sharing that type definition between frontend and dotcom-rendering. Unfortunately we haven't implemented a solution to this. We've implemented the 'validators' such as `getString` to offer basic runtime type checking and the fallbacks when used offer some level of confidence that changes to the model shouldn't cause breaks in production - although you could say we're failing silently here which isn't great for visibility on these issues!